explorer/interpreter/resolve_names.cpp

```diff
@@ -413,6 +413,18 @@ static auto ResolveNames(Statement& statement, StaticScope& enclosing_scope)
       CARBON_RETURN_IF_ERROR(ResolveNames(while_stmt.body(), enclosing_scope));
       break;
     }
+    case StatementKind::For: {
+      StaticScope statement_scope;
+      statement_scope.AddParent(&enclosing_scope);
+      auto& for_stmt = cast<For>(statement);
+      CARBON_RETURN_IF_ERROR(
+          ResolveNames(for_stmt.variable_declaration(), statement_scope));
+      CARBON_RETURN_IF_ERROR(
+          ResolveNames(for_stmt.loop_target(), statement_scope));
```

Comment:
What should happen if the loop variable is mentioned in the loop target? I think that should be an error. Perhaps we could just reverse these two for now, so the loop variable isn't in scope in the loop target.

```suggestion
      CARBON_RETURN_IF_ERROR(
          ResolveNames(for_stmt.loop_target(), statement_scope));
      CARBON_RETURN_IF_ERROR(
          ResolveNames(for_stmt.variable_declaration(), statement_scope));
```